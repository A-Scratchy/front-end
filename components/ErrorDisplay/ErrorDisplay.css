.ErrorDisplay {
  width: 100vw;
  height: 100vh;
  background: #121212;
}

.ErrorDisplay .bg {
  background-image: url("/static/images/TankFlip.gif");
  width: 100vw;
  height: 100vh;
  background-size: cover;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.title {
  font-size: 4rem;
}

.paragraph {
  font-size: 1.4rem;
}
