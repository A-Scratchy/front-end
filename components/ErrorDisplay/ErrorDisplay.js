import React from 'react';
import PropTypes from 'prop-types';
import styles from './ErrorDisplay.css';

ErrorDisplay.propTypes = { statusCode: PropTypes.number.isRequired };

// Formerly known as FourOhFour (404)
function ErrorDisplay({ statusCode }) {
  return (
    <div className={styles.ErrorDisplay}>
      <h1>
        <meta name="robots" content="noindex, nofollow" />
      </h1>
      <div className={styles.bg}>
        <h1 className={styles.title}>{statusCode}!</h1>
        <p className={styles.paragraph}>You definitely weren&apos;t supposed to see this...</p>
      </div>
    </div>
  );
}

export default ErrorDisplay;
