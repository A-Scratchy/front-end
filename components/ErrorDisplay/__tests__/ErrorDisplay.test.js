/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import renderer from "react-test-renderer";

import ErrorDisplay from "../ErrorDisplay";

describe('ErrorDisplay', () => {
    test("should render with just required props", () => {
        createSnapshotTest(
            <ErrorDisplay statusCode="Test Code">
                Test
            </ErrorDisplay>,
        );
    });

    test("should render properly with all props assigned", () => {
        createSnapshotTest(
            <ErrorDisplay
                statusCode="Test Code">
                Test
            </ErrorDisplay>
        );
    });
})
