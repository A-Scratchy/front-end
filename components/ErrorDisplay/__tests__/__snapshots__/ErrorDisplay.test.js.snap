// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay should render properly with all props assigned 1`] = `
<div
  className="ErrorDisplay"
>
  <div
    className="bg"
  >
    <h1
      className="title"
    >
      Test Code
      !
    </h1>
    <p
      className="paragraph"
    >
      You definitely weren't supposed to see this...
    </p>
  </div>
</div>
`;

exports[`ErrorDisplay should render with just required props 1`] = `
<div
  className="ErrorDisplay"
>
  <div
    className="bg"
  >
    <h1
      className="title"
    >
      Test Code
      !
    </h1>
    <p
      className="paragraph"
    >
      You definitely weren't supposed to see this...
    </p>
  </div>
</div>
`;
